test/index.ts

```diff
@@ -54,13 +54,13 @@ describe('tunnel', function() {
 		before(function(done) {
 			this.tunnel = new nodeTunnel.Tunnel();
 			this.events = [];
-			this.tunnel.on('connect', function() {
+			this.tunnel.on('connect', function(this: any) {
```

Comment:
Is `this` within this function the same as `this` outside the function?  It looks to be the case because you use `this.events = []` outside but it's hard to tell.  If that is the case it might be good to do something like
```typescript
type self = typeof this
this.tunnel.on('connect', function(this: self) {
```
(if that's evenpossible), or just use use a fat function like
```typescript
this.tunnel.on('connect', () => {
```
which will make sure `this` is the same for both